sljit_src/sljitProtExecAllocator.c

```diff
@@ -96,51 +102,47 @@ struct chunk_header {
 #endif
 #endif
 
-#if !(defined(__NetBSD__) && defined(MAP_REMAPDUP))
 int mkostemp(char *template, int flags);
 char *secure_getenv(const char *name);
 
 static SLJIT_INLINE int create_tempfile(void)
 {
 	int fd;
-
+	struct stat st;
 	char tmp_name[256];
-	size_t tmp_name_len;
+	size_t tmp_name_len = 0;
 	char *dir;
 	size_t len;
 
-#ifdef P_tmpdir
-	len = (P_tmpdir != NULL) ? strlen(P_tmpdir) : 0;
-
-	if (len > 0 && len < sizeof(tmp_name)) {
-		strcpy(tmp_name, P_tmpdir);
-		tmp_name_len = len;
-	}
-	else {
-		strcpy(tmp_name, "/tmp");
-		tmp_name_len = 4;
-	}
-#else
-	strcpy(tmp_name, "/tmp");
-	tmp_name_len = 4;
-#endif
-
 	dir = secure_getenv("TMPDIR");
 
 	if (dir) {
 		len = strlen(dir);
 		if (len > 0 && len < sizeof(tmp_name)) {
-			strcpy(tmp_name, dir);
+			if ((stat(dir, &st) == 0) && S_ISDIR(st.st_mode)) {
+				strcpy(tmp_name, dir);
+				tmp_name_len = len;
+			}
+		}
+	}
+#ifdef P_tmpdir
+	if (!tmp_name_len) {
+		len = strlen(P_tmpdir);
+		if (len > 0 && len < sizeof(tmp_name)) {
+			strcpy(tmp_name, P_tmpdir);
 			tmp_name_len = len;
 		}
 	}
+#endif
+	if (!tmp_name_len) {
+		strcpy(tmp_name, "/tmp");
+		tmp_name_len = 4;
+	}
 
 	SLJIT_ASSERT(tmp_name_len > 0 && tmp_name_len < sizeof(tmp_name));
 
-	while (tmp_name_len > 0 && tmp_name[tmp_name_len - 1] == '/') {
-		tmp_name_len--;
-		tmp_name[tmp_name_len] = '\0';
-	}
```

Comment:
why is this code needed instead of just removing a trailing '/'?

at least the new version won't convert "////" -> ""